src/models/generativedirectanswer/GenerativeDirectAnswerRequest.ts

```diff
@@ -0,0 +1,16 @@
+import { SearchRequest } from '../core/SearchRequest';
+import { VerticalResults } from '../searchservice/response/VerticalResults';
+
+/**
+ * Options which can be specified for a generative direct answer request.
+ *
+ * @public
+ */
+export interface GenerativeDirectAnswerRequest extends SearchRequest {
+  /** The ID of the search request. */
+  searchId: string,
+  /** The text of the user-written query that prompted Search results. */
+  searchTerm: string,
+  /** The complete set of Search Results */
+  results: VerticalResults | Record<string, VerticalResults[]>
```

Comment:
perhaps I missed something, but why is the type a map of key to results? why not just an array of VerticalResults?

Reply:
we defined results in [GDAApiRequestBody](https://gerrit.yext.com/plugins/gitiles/alpha/+/refs/heads/master/src/com/yext/search/generativeanswers/generative_answers.proto#18) to be of type struct, so we can't pass in array. `VerticalResults` is for a Vertical Search Result, `Record<string, VerticalResults[]>` is for a Universal Search Result. 
I think there is another way I could accomplish this: keep results to be of type `VerticalResults[]`, then in `GDAServiceImpl`, convert request.results to object before sending the api request. Something like this:
```
let results: Object = {}
if (request.results.length === 1) {
    results = request.results[1]
} else if (request.results.length > 1) {
    results = {verticals: request.results}
}
```
let me know if you prefer this way, then I can change and test it

Reply:
yeah, I think that would work better so user of core don't have to transform the universal results themselves